diagnostics: updates to severity_sort

gpanders opened the pull request:
Update the default severity_sort order so that when `severity_sort` is true, higher severities are displayed before lower severities (e.g. ERROR is displayed over WARN).

gpanders committed:
fix(diagnostic): change default severity_sort order

When severity_sort is true, higher severities should be displayed before
lower severities (e.g. ERROR is displayed over WARN).

Also improved the test case for this.

justinmk commented on runtime/doc/diagnostic.txt:
```diff
@@ -239,7 +239,9 @@ config({opts}, {namespace})                          *vim.diagnostic.config()*
-                                   displayed. Options:
+                                   displayed. When true, higher severities are
+                                   displayed before lower severities (e.g.
+                                   ERROR is displayed before WARN). Options:
```
is ERROR (defined as `1`) considered "higher" than WARN (defined as `2`) ? Does this also apply to the min/max filters? so `severity={min=ERROR}` _only_ shows things "more severe" (but lower ordinal value)?

gpanders replied:
Yes that's right. The numbering of the enum items is counter-intuitive (IMO) but is uses to preserve ease of compatibility with the LSP diagnostic severity definitions.

A "higher" severity is a "worse" severity, so ERROR is "greater than" WARN even though their numerical values are the opposite.

gpanders committed:
refactor(diagnostic): group local functions together

gpanders committed:
refactor(diagnostic): combine config() and set() calls

gpanders commented:
@mjlbach I implemented the severity sorting by line as we discussed on Matrix in https://github.com/gpanders/neovim/pull/15701/commits/1f7714763fc24acda60b043406b3502d6c224636, would you mind taking a look when you get a chance?

gpanders commented:
Squeezing in a few other small code cleanup/refactors that don't merit their own PR.

gpanders committed:
fix(diagnostic): remove check on nil return value

vim.diagnostic._set_signs doesn't return anything, so checking the
return value will always fail.
